compiler/src/Transform.js

```diff
@@ -84,30 +105,58 @@ function Transform(
             console.log(this.query);
         }
         this.separable = query["separable"] || true;
+        this.dependencies = [];
         return;
     }
 
-    if (typeof separable !== "boolean")
-        throw new Error("Constructing Transform: separable must be boolean.");
-    if (!Array.isArray(columnNames))
+    // same block of code as above, but for other case!
+    // TODO: remove duplicate code
+    if (Array.isArray(columnNames)) {
```

Comment:
The stuff for updates should be mixed with `columnNames` and should be put in a separate field (e.g. 'updates'). The reasons are that 1) `columnNames` is an array for specifying aliases and 2) putting in a separate field will allow the user to omit the specification of non-updatable fields.

Reply:
Ok, that makes sense. I wanted to avoid having to change every time that a Transform is created in the current visualizations, but I think I can achieve this just using an optional `updateAttrs` parameter, which defaults to an empty object.

Reply:
Exactly. Let me know when you fix this part and I'll move on reviewing other parts of the code. Thanks!